### PR TITLE
Fix UI threading issue + changed lorempixel to static HTTPS image

### PR DESCRIPTION
**Pull request by ahmed-shehata**
Fixed Async issue with UI screens

### CODE REVIEW

File: AtlasDemo/AtlasDemo/ProfileViewController.swift [hunk 1]
```diff
@@ -28,11 +28,11 @@ class ProfileViewController: UIViewController {
...
             switch result {
             case .failure(let error):
-                let alert = UIAlertController(title: "Error", message: "\(error)", preferredStyle: .Alert)
-                self.presentViewController(alert, animated: true, completion: nil)
+                UserMessage.showError(title: "Error", error: error)
 
             case .success(let customer):
-                guard let url = NSURL(string: "http://lorempixel.com/80/80/fashion") else { return }
+                guard let url = NSURL(string:
+                        "https://mir-s3-cdn-cf.behance.net/project_modules/disp/6e9e398680925.560c1799739f9.jpg") else { return }
```
Comment (bonkey) [inline AtlasDemo/AtlasDemo/ProfileViewController.swift]: Hmmm. Maybe static file and then we can kill Haneke and make deps building faster?

Comment (bonkey) [inline AtlasDemo/AtlasDemo/ProfileViewController.swift]: Will we get any user's profile?

File: AtlasSDK/AtlasSDK/APIRequestBuilder.swift [hunk 2]
```diff
@@ -18,16 +18,20 @@ final class APIRequestBuilder: RequestBuilder {
             switch result {
             case .failure(let error):
                 if let error = error as? AtlasAPIError where error.code == HTTPStatus.Unauthorized {
-                    guard let strongSelf = self else { return }
-                    strongSelf.loginAndExecute(completion)
+                    dispatch_async(dispatch_get_main_queue()) { [weak self] in
```
Comment (bonkey) [inline AtlasSDK/AtlasSDK/APIRequestBuilder.swift]: If Async.main {} helped in AtlasUI, don't force it here.

File: AtlasSDK/AtlasSDK/APIRequestBuilder.swift [hunk 3]
```diff
@@ -18,16 +18,20 @@ final class APIRequestBuilder: RequestBuilder {
...
+                    }
                 } else {
                     completion(.failure(error))
                     strongSelf.executionFinished?(strongSelf)
                 }
             case .success(let response):
-                guard let strongSelf = self else { return }
-                completion(.success(response))
-                strongSelf.executionFinished?(strongSelf)
+                dispatch_async(dispatch_get_main_queue()) { [weak self] in
```
Comment (bonkey) [inline AtlasSDK/AtlasSDK/APIRequestBuilder.swift]: If Async.main {} helped in AtlasUI, don't force it here.
 Add a line note

File: AtlasSDK/AtlasSDK/de.lproj/Localizable.strings [hunk 4]
```diff
@@ -9,4 +9,6 @@
 "Total" = "Gesamt";
 "Shipping" = "Versand";
 "Pay %@" = "Zahlen %@";
-"Connect to Zalando" = "Mit Zalando verbinden";
\ No newline at end of file
+"Connect to Zalando" = "Mit Zalando verbinden";
+
+"CheckoutSummaryViewController.terms" = "Zalando terms, conditions and cancellation policy apply to all orders";
```
Comment (bonkey) [inline AtlasSDK/AtlasSDK/de.lproj/Localizable.strings]: Please make all other keys in same syntax as well.

File: AtlasSDK/AtlasSDK/en.lproj/Localizable.strings [hunk 5]
```diff
@@ -8,4 +8,8 @@
 "Done" = "Done";
 "Total" = "Total";
 "Shipping" = "Shipping";
-"Pay %@" = "Pay %@";
\ No newline at end of file
+"Pay %@" = "Pay %@";
+
+"CheckoutSummaryViewController.terms" = "Zalando terms, conditions and cancellation policy apply to all orders";
+
+"Zalando terms, conditions and cancellation policy apply to all orders" = "Zalando terms, conditions and cancellation policy apply to all orders";
```
Comment (bonkey) [inline AtlasSDK/AtlasSDK/en.lproj/Localizable.strings]: What's this? 👹

File: AtlasUI/AtlasUI/CheckoutSummaryViewController.swift [hunk 6]
```diff
@@ -31,24 +31,23 @@ final class CheckoutSummaryViewController: UIViewController {
...
-        self.view.backgroundColor = UIColor.clearColor()
-        self.view.opaque = false
-        self.setupNavBar()
-        self.setupBlurView()
-        self.setupProductImageView()
-        self.setupViewLabels()
-        self.setupStackView()
-        self.setupTermsAndConditionsButton()
-        self.setupBuyButton()
+        Async.main {
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryViewController.swift]: `viewDidLoad` runs on main, so it's redundant.

File: AtlasUI/AtlasUI/CheckoutSummaryViewController.swift [hunk 7]
```diff
@@ -66,15 +65,18 @@ final class CheckoutSummaryViewController: UIViewController {
...
-        self.view.addSubview(backgroundView)
-        self.view.addSubview(indicator)
+        Async.main {
+            let indicator: UIActivityIndicatorView = UIActivityIndicatorView(activityIndicatorStyle: .WhiteLarge)
+            let backgroundView = UIView(frame: CGRect(x: 0, y: 0, width: self.view.bounds.size.width, height: self.view.bounds.size.height))
+            backgroundView.backgroundColor = .grayColor()
+            backgroundView.alpha = 0.2
+            indicator.center = CGPoint(x: self.view.bounds.size.width / 2, y: (self.view.bounds.size.height) / 2)
+            indicator.color = .blackColor()
+            indicator.startAnimating()
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryViewController.swift]: Make it the last line in this func and only this embrace with Async.main {}

File: AtlasUI/AtlasUI/CheckoutSummaryViewController.swift [hunk 8]
```diff
@@ -91,15 +93,18 @@ final class CheckoutSummaryViewController: UIViewController {
...
     private func connectToZalando() {
         AtlasSDK.fetchCustomer { result in
-            switch result {
-            case .failure(let error):
-                let alert = UIAlertController(title: "Error".loc, message: "\(error)", preferredStyle: .Alert)
-                self.presentViewController(alert, animated: true, completion: nil)
+            Async.main {
+                switch result {
+                case .failure(let error):
+                    let alert = UIAlertController(title: "Error".loc, message: "\(error)", preferredStyle: .Alert)
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryViewController.swift]: Use UserMessage.

File: AtlasUI/AtlasUI/CheckoutSummaryViewController.swift [hunk 9]
```diff
@@ -91,15 +93,18 @@ final class CheckoutSummaryViewController: UIViewController {
...
+                case .success(let customer):
+                    self.showLoadingView()
 
-            case .success(let customer):
-                self.showLoadingView()
+                    guard let article = self.checkoutViewModel.article,
+                        articleIndex = self.checkoutViewModel.articleUnitIndex else { return }
 
-                if let article = self.checkoutViewModel.article, articleIndex = self.checkoutViewModel.articleUnitIndex {
                     self.checkoutService.generateCheckout(withArticle: article, articleUnitIndex: articleIndex) { result in
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryViewController.swift]: Don't forget about separated func for this

File: AtlasSDK/AtlasSDK/APIRequestBuilder.swift (hunk 2 above)
Comment (ahmed-shehata) [inline AtlasSDK/AtlasSDK/APIRequestBuilder.swift]: Unfortunately it did not

File: AtlasDemo/AtlasDemo/ProfileViewController.swift [hunk 10]
```diff
@@ -31,9 +31,7 @@ class ProfileViewController: UIViewController {
                 UserMessage.showError(title: "Error", error: error)
 
             case .success(let customer):
-                guard let url = NSURL(string:
-                        "https://mir-s3-cdn-cf.behance.net/project_modules/disp/6e9e398680925.560c1799739f9.jpg") else { return }
-                self.avatar.hnk_setImageFromURL(url)
+                self.avatar.image = UIImage(named: "user")
```
Comment (bonkey) [inline AtlasDemo/AtlasDemo/ProfileViewController.swift]: And kill Haneke from `Cartfile` please

File: AtlasUI/AtlasUI/CheckoutSummaryViewController.swift (hunk 6 above)
Comment (ahmed-shehata) [inline AtlasUI/AtlasUI/CheckoutSummaryViewController.swift]: setupViews is called elsewhere, so it is needed

**Pull request merged**